opm/autodiff/BlackoilModelBase.hpp

```diff
@@ -175,7 +176,7 @@ namespace Opm {
         /// \param[in, out] reservoir_state   reservoir state variables
         /// \param[in, out] well_state        well state variables
         template <class NonlinearSolverType>
-        IterationReport nonlinearIteration(const int iteration,
+        SimulatorReport nonlinearIteration(const int iteration,
```

Comment:
Perhaps we are not quite aligned as to how we want to use this member, but for sure: we must stop throwing if we are to propagate such info back to the caller. I am not proposing to do that with this PR though!